lxd/cluster/membership.go

```diff
@@ -847,8 +847,28 @@ func Assign(state *state.State, gateway *Gateway, nodes []db.RaftNode) error {
 
 		// If we're stepping back from voter to spare, let's first transition
 		// to stand-by first and wait for the configuration change to be
-		// notified to us. This prevent us from thinking we're still voters and
-		// potentially disrupt the cluster.
+		// notified to us. This prevents us from thinking we're still a voter and
+		// potentially disrupting the cluster.
+		//
+		// If we went from voter -> spare directly:
```

Comment:
Excellent detail, thanks!

Reply:
Although its unfortunate go-dqlite doesn't prevent this sort of thing internally and requires each application to know these fine details.